src/Database/Dialect/SqlserverDialectTrait.php

```diff
@@ -102,7 +104,31 @@ public function _version()
     protected function _pagingSubquery($original, $limit, $offset)
     {
         $field = '_cake_paging_._cake_page_rownum_';
-        $order = $original->clause('order') ?: new OrderByExpression('(SELECT NULL)');
+
+        if ($original->clause('order')) {
+            $order = clone $original->clause('order');
+        } else {
+            $order = new OrderByExpression('(SELECT NULL)');
+        }
+
+        // SQL server does not support column aliases in OVER clauses.  But for
+        // calculated columns the alias is the only practical identifier to use
+        // when specifying the order.  So if a column alias is specified in the
+        // order clause, and the value of that alias is an expression, change
+        // the alias into what it represents by setting the clause's key to be
+        // the SQL representation of its value.  The UnaryExpression creation
+        // below will then do the right thing and use the calculation in the
+        // ROW_NUMBER() OVER clause instead of the alias.
+        $select = $original->clause('select');
+        $order->iterateParts(function ($direction, &$orderBy) use ($select) {
+            if (isset($select[$orderBy])) {
+                if ($select[$orderBy] instanceof ExpressionInterface) {
+                    $orderBy = $select[$orderBy]->sql(new ValueBinder());
```

Comment:
Why do you need the bound parameters? Does attaching the expression to the orderby clause not work?